### ISSUE TITLE
Fine-tune optimal hyperparameters for small datasets?

### ISSUE THREAD

**Issue by adiIspas** [NVIDIA/NeMo#3226]
I'm trying to improve the accuracy of the ASR model for a specific domain.

The base model it's a fine-tuned model from English (QuartzNet) to my language, and now I'm trying to improve this base model on a specific dataset.

The dataset has around 12h, and I used speed perturbation to generate around 37h.

I use these hyperparameters by the loss seems to stay the same after a significant number of epochs.

```
new_optimizer = {
        'name': 'novograd',
        'lr': 0.0001,
        'betas': (0.95, 0.25),
        'weight_decay': 0.001,
        'sched': {
            'last_epoch': -1,
            'min_lr': 0.0,
            'monitor': 'val_loss',
            'name': 'CosineAnnealing',
            'reduce_on_plateau': False,
            'warmup_ratio': 0.12,
            'warmup_steps': None,
        },
    }
```

I use 500 epochs for this training.

Do you have any advice on how I can optimally improve the accuracy of this small dataset, or how is a minimum number of hours that can generate a good improvement?

**Comment by titu1994** [comment]
The LR is too low. You should use 1/5 or 1/10th the original LR - that's around 0.005 to 0.001 for QuartzNet. 

Your dataset is also too small. Refer to the finetuning for different language notebook on how to freeze your encoder and fine-tune just your decoder and batch norm layers.

**Comment by adiIspas** [comment]
Hi @titu1994 thanks for your reply!

I already performed a transfer learning from English to my language for the baseline model (let's say RoModel), for which I used over 200h and 0.001 as the learning rate.

Now, I tried to improve the RoModel with transfer learning for domain-specific data in the same language, where I have around 12h, and I use 1/10th from the initial learning rate, 0.0001.

It's that learning rate right or do I have to use 0.001 for transfer learning from RoModel to domain-specific?

**Comment by titu1994** [comment]
0.001 is fine for both runs. Still, ensure that you freeze the encoder.

**Comment by adiIspas** [comment]
Ok, cool. Thanks for your advice @titu1994. I'll try with 0.001 as learning rate for this domain fine tune.

**Comment by adiIspas** [comment]
Another short question @titu1994.

Batch normalization has to been unfrozen if the language remains the same, and I perform some specific domain fine-tune?

**Comment by titu1994** [comment]
If you dataset is so small, it's best to freeze the encoder then unfreeze the batch norm. Irrespective of same or different language.
